### DISCUSSION TITLE
Providing trivy with information about packages not installed from OS package ecosystem

### DISCUSSION THREAD

**Discussion by loosebazooka** [aquasecurity/trivy#5220] [Ideas]
### Description

Distroless uses *mostly* debian packages except nodejs. We mistakenly provide images for our node packages (that come directly from nodejs.org) as debian metadata. This is causing some confusion to scanners as they go looking to the debian security database for the node install when no matching information exists. While we have an sbom, I think it would still be nice for trivy and potentially other scanners to provide a way for us to report packages not coming from OS distributions (unless this already exists and I'm mistaken)

### Target

Container Image

### Scanner

Vulnerability

**Comment by itaysk** [comment]
you mention that you do have an sbom, would scanning the SBOM with Trivy solve your problem (`trivy sbom`)? Trivy can also automatically find the SBOM alongside the image in a registry (`trivy image --sbom-sources oci`).

**Reply by loosebazooka** [reply]
Gotcha. Lemme try that and see if that's the recommendation we should make to our users.
